src/core/Akka.Remote/EndpointManager.cs

```diff
@@ -492,196 +543,248 @@ protected void Accepting(object message)
+                    .With<Pass>(pass =>
+                    {
+                        pass.Endpoint.Tell(new ReliableDeliverySupervisor.Ungate());
+                    });
             }
             else
             {
-                if (endpoints.IsQuarantined(handle.RemoteAddress, (int)handle.HandshakeInfo.Uid))
+                if (_endpoints.IsQuarantined(handle.RemoteAddress, handle.HandshakeInfo.Uid))
                     handle.Disassociate(DisassociateInfo.Quarantined);
                 else
                 {
-                    if (endpoints.WritableEndpointWithPolicyFor(handle.RemoteAddress) is Pass)
+                    var policy = _endpoints.WritableEndpointWithPolicyFor(handle.RemoteAddress);
+                    var pass = policy as Pass;
+                    if (pass != null && !pass.Uid.HasValue)
                     {
-                        var pass = (Pass) endpoints.WritableEndpointWithPolicyFor(handle.RemoteAddress);
-                        if (!pass.Uid.HasValue)
+                        // Idle writer will never send a GotUid or a Terminated so we need to "provoke it"
+                        // to get an unstash event
+                        if (!writerIsIdle)
                         {
-                            if (stashedInbound.ContainsKey(pass.Endpoint)) stashedInbound[pass.Endpoint].Add(ia);
-                            else stashedInbound.AddOrSet(pass.Endpoint, new List<InboundAssociation>() {ia});
+                            pass.Endpoint.Tell(ReliableDeliverySupervisor.IsIdle.Instance);
+                            var stashedInboundForEp = _stashedInbound.GetOrElse(pass.Endpoint,
+                                new List<InboundAssociation>());
+                            stashedInboundForEp.Add(ia);
+                            _stashedInbound[pass.Endpoint] = stashedInboundForEp;
                         }
                         else
                         {
-                            if (handle.HandshakeInfo.Uid == pass.Uid)
-                            {
-                                if (pendingReadHandoffs.ContainsKey(pass.Endpoint))
-                                    pendingReadHandoffs[pass.Endpoint].Disassociate();
-                                pendingReadHandoffs.AddOrSet(pass.Endpoint, handle);
-                                pass.Endpoint.Tell(new EndpointWriter.StopReading(pass.Endpoint, Self));
-                            }
-                            else
-                            {
-                                Context.Stop(pass.Endpoint);
-                                endpoints.UnregisterEndpoint(pass.Endpoint);
-                                pendingReadHandoffs.Remove(pass.Endpoint);
-                                CreateAndRegisterEndpoint(handle, pass.Uid);
-                            }
+                            CreateAndRegisterEndpoint(handle, _endpoints.RefuseUid(handle.RemoteAddress));
```

Comment:
If the writer is not idle, create and register a new endpoint and get the refused UID of the remote address